app/scripts/modules/core/delivery/filter/executionFilter.service.ts

```diff
@@ -0,0 +1,320 @@
+
+import {ILogService, module} from 'angular';
+import {chain, compact, debounce, find, flattenDeep, forOwn, get, groupBy, includes, uniq} from 'lodash';
+
+import {Application} from 'core/application/application.model';
+import {EXECUTION_FILTER_MODEL, ExecutionFilterModel} from 'core/delivery/filter/executionFilter.model';
+import {IExecution} from 'core/domain/IExecution';
+import {IPipeline} from 'core/domain/IPipeline';
+
+interface IGroup {
+  heading: string;
+  executions: IExecution[];
+  config?: any;
+  runningExecutions?: IExecution[];
+  targetAccounts?: string[];
+}
+
+export class ExecutionFilterService {
+  private lastApplication: Application = null;
+  private isFilterable: (sortFilterModel: any[]) => boolean;
+
+  // this gets called every time the URL changes, so we debounce it a tiny bit
+  public updateExecutionGroups: (application: Application) => IGroup[];
+
+
+
+  static get $inject(): string[] { return ['executionFilterModel', 'timeBoundaries', 'waypointService', '$log', 'filterModelService', 'pipelineConfig']; }
+
+  constructor(private executionFilterModel: ExecutionFilterModel,
+              private timeBoundaries: any,
+              waypointService: any,
```

Comment:
Why no access modifier on `waypointService`? It doesn't need to be public, does it?